src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -18,10 +18,11 @@
 az_config = CLIConfig(config_dir=GLOBAL_CONFIG_DIR, config_env_var_prefix=ENV_VAR_PREFIX)
 _CUSTOM_EXT_DIR = az_config.get('extension', 'dir', None)
 _DEV_EXTENSION_SOURCES = az_config.get('extension', 'dev_sources', None)
+_CUSTOM_EXT_SYS_DIR = az_config.get('extension', 'sys_dir', None)
 EXTENSIONS_DIR = os.path.expanduser(_CUSTOM_EXT_DIR) if _CUSTOM_EXT_DIR else os.path.join(GLOBAL_CONFIG_DIR,
                                                                                           'cliextensions')
 DEV_EXTENSION_SOURCES = _DEV_EXTENSION_SOURCES.split(',') if _DEV_EXTENSION_SOURCES else []
-EXTENSIONS_SYS_DIR = os.path.join(get_python_lib(), 'azure-cli-extensions') if sys.platform.startswith('linux') else ""
+EXTENSIONS_SYS_DIR = os.path.expanduser(_CUSTOM_EXT_SYS_DIR) if _CUSTOM_EXT_SYS_DIR else os.path.join(get_python_lib(), 'azure-cli-extensions')
```

Comment:
Need to test if this also works on Windows and Mac.

Reply:
On windows, users need to open the shell as administrator. I have added [message](https://github.com/Azure/azure-cli/pull/12856/files#diff-c9aa0ae7daec922ec8ebbd390137a745R59) when users encounter the permission error.

Reply:
run as admin only required when user set customized install dir right? if yes, then less concern since no change to existing behavoir

Reply:
yes, not affecting default extension installation.